java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/AdminController.java

```diff
@@ -492,7 +436,22 @@ public String index(Model model) throws SQLException {
 		return "admin/info";
 	}
 	
-
+	private Map<String, String> getLastUsersPromo(List<PromoRepository.Promo> promos) {
```

Comment:
Method is too slow, will be an issue once it will be growing

Reply:
Just to remind it's a for loop of for loops SQL query, this is definitely not good. It should be 1 query or probably if it is used for information only, it's ok to store inside promocampaign table